crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,759 @@
+                false,
+                Nonce::sample(),
+                PublicKey::sample(),
+            )
+            .await;
+
+        assert_eq!(
+            result,
+            Err(CommonError::ExecutionSummaryFail {
+                underlying: "InvalidReceipt".to_string()
+            })
+        )
+    }
+
+    #[actix_rt::test]
+    async fn execution_summary_reserved_instructions_error() {
+        let zero = ScryptoDecimal192::zero();
+        let responses = prepare_responses(
+            LedgerState {
+                network: "".to_string(),
+                state_version: 0,
+                proposer_round_timestamp: "".to_string(),
+                epoch: 0,
+                round: 0,
+            },
+            TransactionPreviewResponse {
+                encoded_receipt: "".to_string(),
+                radix_engine_toolkit_receipt: Some(ScryptoSerializableToolkitTransactionReceipt::CommitSuccess {
+                    state_updates_summary: native_radix_engine_toolkit::receipt::StateUpdatesSummary {
+                        new_entities: IndexSet::new(),
+                        metadata_updates: IndexMap::new(),
+                        non_fungible_data_updates: IndexMap::new(),
+                        newly_minted_non_fungibles: IndexSet::new(),
+                    },
+                    worktop_changes: IndexMap::new(),
+                    fee_summary: native_radix_engine_toolkit::receipt::FeeSummary {
+                        execution_fees_in_xrd: zero.into(),
+                        finalization_fees_in_xrd: zero.into(),
+                        storage_fees_in_xrd: zero.into(),
+                        royalty_fees_in_xrd: zero.into(),
+                    },
+                    locked_fees: native_radix_engine_toolkit::receipt::LockedFees {
+                        contingent: zero.into(),
+                        non_contingent: zero.into(),
+                    },
+                }),
+                logs: vec![],
+                receipt: TransactionReceipt {
+                    status: TransactionReceiptStatus::Succeeded,
+                    error_message: None,
+                },
+            },
+        );
+        let os =
+            prepare_os(MockNetworkingDriver::new_with_bodies(200, responses))
+                .await;
+
+        let result = os
+            .perform_transaction_preview_analysis(
+                prepare_manifest_with_account_entity().instructions_string(),
+                Blobs::sample(),
+                Message::sample(),
+                false,
+                Nonce::sample(),
+                PublicKey::sample(),
+            )
+            .await;
+
+        assert_eq!(
+            result,
+            Err(CommonError::ReservedInstructionsNotAllowedInManifest {
+                reserved_instructions: "AccountLockFee".to_string()
+            })
+        )
+    }
+
+    #[actix_rt::test]
+    async fn success() {
+        let responses = prepare_responses(
```

Comment:
same here.. with `zero`. perhaps put the `let zero = RETDecimal::ZERO;` inside the `mod tests` and let it be used by all tests.